File: backend/app/schemas/statistics.py
from datetime import date

from pydantic import BaseModel


class DailyStatsSummary(BaseModel):
    date: date
    calories: int
    protein: float
    fat: float
    carbs: float
    goal: int | None = None
    entries_count: int


class WeightDataPoint(BaseModel):
    date: date
    weight: float


class MacroAverages(BaseModel):
    calories: float
    protein: float
    fat: float
    carbs: float


class StatisticsResponse(BaseModel):
    period_start: date
    period_end: date
    daily_stats: list[DailyStatsSummary]
    weight_data: list[WeightDataPoint]
    averages: MacroAverages
    total_entries: int
    days_with_entries: int

File: frontend/src/types/statistics.ts
export interface DailyStatsSummary {
  date: string;
  calories: number;
  protein: number;
  fat: number;
  carbs: number;
  goal: number | null;
  entries_count: number;
}

export interface WeightDataPoint {
  date: string;
  weight: number;
}

export interface MacroAverages {
  calories: number;
  protein: number;
  fat: number;
  carbs: number;
}

export interface StatisticsResponse {
  period_start: string;
  period_end: string;
  daily_stats: DailyStatsSummary[];
  weight_data: WeightDataPoint[];
  averages: MacroAverages;
  total_entries: number;
  days_with_entries: number;
}

export type StatisticsPeriod = 'week' | 'month';
